packages/react/src/components/Cta/CTA.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { ArrowDown20, ArrowRight20, Launch20 } from '@carbon/icons-react';
+import { ButtonGroup } from '../../patterns/sub-patterns/ButtonGroup';
+import { CardLink } from '../../patterns/sub-patterns/CardLink';
+import { FeaturedLink } from '../../patterns/blocks/FeaturedLink';
+import { LinkWithIcon } from '../LinkWithIcon';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { smoothScrolling } from '@carbon/ibmdotcom-utilities';
+/**
+ * CTA component
+ *
+ * @param {object} props props object
+ * @param {string} props.style style ( text | card | button | feature ).
+ * @param {object} props.type tyle ( jump | local | external ).
+ * @param {object} props.cta cta object which includes href,text, copy, card and image, button  properties.
+ * @returns {*} CTA Component
+ */
+const CTA = ({ style, type, ...cta }) => {
+  switch (style) {
+    case 'card':
+      return (
+        <CardLink
+          {...cta}
+          icon={iconSelector(type)}
+          onClick={type === 'jump' ? smoothScrolling : null}
+        />
+      );
+    case 'button':
+      return (
+        <ButtonGroup
+          buttons={renderButtons(cta)}
+          onClick={type === 'jump' ? smoothScrolling : null}
+        />
+      );
+    case 'feature':
+      return <FeaturedLink {...cta} />;
+    default:
+      return (
+        <LinkWithIcon
+          href={cta.href}
+          target={type === 'external' ? '_blank' : null}
```

Comment:
This should be applied to the other styles I believe. I wonder if it makes sense to pull this logic out into a separate function so we don't have to repeat all of this logic 3 times.

```javascript
<CardLink
  {...cta}
  icon={iconSelector(type)}
  target={external()}
  onClick={type === 'jump' ? smoothScrolling : null}
/>

function external (type) {
  return type === 'external' ? '_blank' : null;
}
```

I guess you could even do this with the `onClick` as well, that way the logic only has to be updated once. I keep going back and forth on this one. It's almost easier to just write the `onClick` out on its own because you have to pass in all the props and execute an additional two functions.


```javascript
<CardLink
  {...cta}
  icon={iconSelector(type)}
  target={external()}
  onClick={e => { jump(e, type) }}
/>

function jump (e, type) {
  return type === 'jump' ? SmoothScroll(e) : null;
}
```

Reply:
Sure.